components/org.wso2.carbon.identity.api.server.branding.preference.management/org.wso2.carbon.identity.api.server.branding.preference.management.v1/src/main/java/org/wso2/carbon/identity/api/server/branding/preference/management/v1/impl/BrandingPreferenceApiServiceImpl.java

```diff
@@ -0,0 +1,121 @@
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.wso2.carbon.identity.api.server.branding.preference.management.v1.BrandingPreferenceApiService;
+import org.wso2.carbon.identity.api.server.branding.preference.management.v1.core.BrandingPreferenceManagementService;
+import org.wso2.carbon.identity.api.server.branding.preference.management.v1.model.BrandingPreferenceModel;
+import org.wso2.carbon.identity.api.server.common.ContextLoader;
+
+import java.net.URI;
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.identity.api.server.branding.preference.management.common.BrandingPreferenceManagementConstants.APPLICATION_TYPE;
+import static org.wso2.carbon.identity.api.server.branding.preference.management.common.BrandingPreferenceManagementConstants.BRANDING_PREFERENCE_CONTEXT_PATH;
+import static org.wso2.carbon.identity.api.server.branding.preference.management.common.BrandingPreferenceManagementConstants.CUSTOM_TYPE;
+import static org.wso2.carbon.identity.api.server.branding.preference.management.common.BrandingPreferenceManagementConstants.DEFAULT_LOCALE;
+import static org.wso2.carbon.identity.api.server.branding.preference.management.common.BrandingPreferenceManagementConstants.GET_PREFERENCE_COMPONENT_WITH_QUERY_PARAM;
+import static org.wso2.carbon.identity.api.server.branding.preference.management.common.BrandingPreferenceManagementConstants.ORGANIZATION_TYPE;
+import static org.wso2.carbon.identity.api.server.common.Constants.V1_API_PATH_COMPONENT;
+import static org.wso2.carbon.identity.api.server.common.ContextLoader.getTenantDomainFromContext;
+
+/**
+ * Implementation of branding preference management REST API.
+ */
+public class BrandingPreferenceApiServiceImpl implements BrandingPreferenceApiService {
+
+    @Autowired
+    private BrandingPreferenceManagementService brandingPreferenceManagementService;
+
+    //TODO: Improve API to manage application level & language level theming resources in addition to the tenant level.
+
+    @Override
+    public Response addBrandingPreference(BrandingPreferenceModel brandingPreferenceModel) {
+
+        if (StringUtils.isBlank(brandingPreferenceModel.getType().toString())) {
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+        if (!StringUtils.equals(brandingPreferenceModel.getType().toString(), ORGANIZATION_TYPE)) {
+            return Response.status(Response.Status.NOT_IMPLEMENTED).build();
+        }
+        if ((!StringUtils.equals(brandingPreferenceModel.getLocale(), DEFAULT_LOCALE)) &&
+                StringUtils.isNotBlank(brandingPreferenceModel.getLocale())) {
+            return Response.status(Response.Status.NOT_IMPLEMENTED).build();
+        }
+
+        brandingPreferenceManagementService.addBrandingPreference(brandingPreferenceModel);
+        URI location = ContextLoader.buildURIForHeader(V1_API_PATH_COMPONENT + BRANDING_PREFERENCE_CONTEXT_PATH
```

Comment:
Let's URL encode the query params

Reply:
Addressed with https://github.com/wso2/identity-api-server/pull/332/commits/95afc9eceedc9e6e0e0e8eec8bb23634936f37c1